pipeline/mutate/mutator_id_token.go

```diff
@@ -204,5 +204,29 @@ func (a *MutatorIDToken) Config(config json.RawMessage) (*CredentialsIDTokenConf
 		c.TTL = "15m"
 	}
 
+	if a.tokenCache == nil {
+		cost := int64(c.Cache.MaxCost)
+		if cost == 0 {
+			cost = 1 << 25
+		}
+
+		cache, err := ristretto.NewCache(&ristretto.Config{
+			NumCounters: cost * 10,
+			// Allocate a max
+			MaxCost: cost,
+			// This is a best-practice value.
+			BufferItems: 64,
+			Cost: func(value interface{}) int64 {
+				return 1
+			},
+			IgnoreInternalCost: true,
```

Comment:
To be discussed if this is really the best strategy for `id_token`.

I'm less familiar with other use cases of generated `id_token` and also how `ristretto` computes the "cost".

For now, this is copied from the AuthN OAuth2 introspection handler.
But perhaps saying that each id_token = 1 cost would cache too many id_tokens.

I'd like @aeneasr's opinion here :)

Reply:
In the current state it does not work, but we could at least do
```
                        NumCounters: 10000,
			BufferItems: 64,
                        MaxCost: cost,
```
to keep the same behaviour as previously.

Reply:
cost = 1 << 25

as the default is definitely too many now that the cost function is 1

Reply:
Since JWTs can be quite long it may make more sense to calculate the cost based on the string length to have an approximation of storage use

Reply:
That could make sense indeed 👍 
Then the default limit of 1 << 25 (~33.5 million), if we estimate an average length of JWT of a few thousands char, we should be able to store some tens of thousands of keys - which sounds reasonable.